tools/login/actions/login.php

```diff
@@ -113,8 +126,17 @@
 if ($_REQUEST["action"] == "logout") {
     $this->LogoutUser();
     $this->SetMessage(_t('LOGIN_YOU_ARE_NOW_DISCONNECTED'));
-    $this->Redirect(preg_replace('/(&|\\\?)$/m', '', preg_replace('/(&|\\\?)action=logout(&)?/', '$1', $incomingurl)));
-    $this->exit();
+    
+    if (isset($loggedouturl))
+    {
+    	$this->Redirect(preg_replace('/(&|\\\?)$/m', '', preg_replace('/(&|\\\?)action=logout(&)?/', '$1', $incomingurl)));
+    	$this->exit();
+    }
+    else
```

Comment:
```suggestion
    else if ($this->IsWikiName($loggedouturl, WN_CAMEL_CASE_EVOLVED))
```
serait un peu plus sûr pour éviter les erreurs avec des urls qui ne sont pas des Wikiname

Reply:
Je ne sais pas dans quelle mesure cela est gênant de spécifier une url qui n'est pas un wikiname... Si c'est genant il faudrait dans ce cas tester les autres urls comme loggedinurl.